crates/sdk/src/error.rs

```diff
@@ -0,0 +1,96 @@
+use std::sync::Arc;
+
+use thiserror::Error;
+
+#[derive(Error, Debug, Clone)]
+#[non_exhaustive]
+pub enum Error {
+    #[error("Disconnected normally following a call to `DbContext::disconnect`")]
+    Disconnected,
```

Comment:
It looks like there are 3 errors that effectively mean we are disconnected (`Disconnected`, `AlreadyDisconnected`, and `Ws`). Can we combine these three? Maybe `Disconnected` with an optional underlying error?

Reply:
I agree that `Disconnected` and `AlreadyDisconnected` are redundant, and am happy to combine them. I think I should restructure `Ws` so that we can distinguish between failed-to-connect versus erroneous-disconnect-after-the-fact, since I believe it can occur in both of those cases. I am unsure whether erroneous-disconnect should be the same error as disconnected-on-purpose; my first intuition was no, but if you think that's wrong, I'm happy to combine the two categories, s.t. `Disconnected { source: None }` means normal disconnect and  `Disconnected { source: Some(err) }` means abnormal disconnect.